serving/src/main/java/ai/djl/serving/ModelServer.java

```diff
@@ -415,6 +415,10 @@ private String inferEngineFromUrl(String modelUrl) {
         try {
             Repository repository = Repository.newInstance("modelStore", modelUrl);
             List<MRL> mrls = repository.getResources();
+            if (mrls.isEmpty()) {
+                throw new AssertionError("Model directory path does not exist : " + modelUrl);
```

Comment:
```suggestion
                throw new IllegalArgumentException("Invalid model url: " + modelUrl);
```

Reply:
AssertionError should only be used for the case that it should never happen.

Reply:
I see. Okay!